### PR TITLE
Add explanations regarding actix-web's multithreaded server model

### PR DESCRIPTION
**Pull request by vmalloc**

### CODE REVIEW

File: content/docs/server.md [hunk 1]
```diff
@@ -31,18 +31,48 @@ Following example shows how to start http server in separate thread.
...
+Once the workers are created, they each receive a separate *application* instance to handle
+requests. Application state is not shared between the threads, and handlers are free to manipulate
+their data with no concurrency concerns.
 
 > Application state does not need to be `Send` or `Sync`, but application
 factory must be `Send` + `Sync`.
 
+If one wishes to share state between worker threads, `Arc` can be used. Special attention should be
+given once such sharing is introduced, because in many cases performance costs might be
+inadvertently introduced as a result of locking the shared state for modifications.
```
Comment (robjtede) [inline content/docs/server.md]: A bit more concise and direct.

> To share state between worker threads, use an `Arc`. Special care should be taken once sharing and synchronization is introduced. In many cases, performance costs are inadvertently introduced as a result of locking the shared state for modifications.

May also be worth elaborating on this (or pointing to a resource) or talking about using `Mutex` vs `RwLock` vs some [futures-based locking primitives](https://docs.rs/futures/0.3.1/futures/lock/struct.Mutex.html).

File: content/docs/server.md [hunk 2]
```diff
@@ -31,18 +31,48 @@ Following example shows how to start http server in separate thread.
...
 equal to number of logical CPUs in the system. This number can be overridden with the
 [`HttpServer::workers()`][workers] method.
 
 {{< include-example example="server" file="workers.rs" section="workers" >}}
 
-The server creates a separate application instance for each created worker. Application state
-is not shared between threads. To share state, `Arc` could be used.
+Once the workers are created, they each receive a separate *application* instance to handle
+requests. Application state is not shared between the threads, and handlers are free to manipulate
+their data with no concurrency concerns.
```
Comment (robjtede) [inline content/docs/server.md]:
```suggestion
their copy of the state with no concurrency concerns.
```

File: content/docs/server.md (hunk 1 above)
Comment (vmalloc) [reply content/docs/server.md]: @robjtede thanks for the feedback! I pushed another revision in which I tried to tackle this point. Let me know what you think

File: content/docs/server.md [hunk 3]
```diff
@@ -31,18 +31,53 @@ Following example shows how to start http server in separate thread.
...
-is not shared between threads. To share state, `Arc` could be used.
+Once the workers are created, they each receive a separate *application* instance to handle
+requests. Application state is not shared between the threads, and handlers are free to manipulate
+their copy of the state with no concurrency concerns.
 
 > Application state does not need to be `Send` or `Sync`, but application
 factory must be `Send` + `Sync`.
 
+To share state between worker threads, use an `Arc`. Special care should be taken once sharing and
+synchronization is introduced. In many cases, performance costs are inadvertently introduced as a
```
Comment (JohnTitor) [inline content/docs/server.md]:
```suggestion
synchronization are introduced. In many cases, performance costs are inadvertently introduced as a
```

**Comment by vmalloc** [comment]
@JohnTitor fixed! Any more feedback before I merge this?

**Comment by JohnTitor** [comment]
I believe there is nothing to do, thanks!

**Comment by vmalloc** [comment]
You're most welcome. Thanks for the feedback!

File: content/docs/server.md [hunk 4]
```diff
@@ -31,18 +31,53 @@ Following example shows how to start http server in separate thread.
...
+
+Since each worker thread processes its requests sequentially, handlers which block the current
+thread will cause the current worker to stop processing new requests:
+
+```rust
+fn my_handler() -> impl Responder {
+    std::thread::sleep(Duration::from_secs(5)); // <-- Bad practice! Will cause the current worker thread to hang!
+    "response"
+}
+```
```
Comment (cdbattags) [inline content/docs/server.md]: by chance can we get a line break here?

Comment (vmalloc) [reply content/docs/server.md]: Done!

**Pull request merged**